python/ray/util/tracing/tracing_helper.py

```diff
@@ -325,9 +329,6 @@ def _inject_tracing_into_function(function):
     Use the provided trace context from kwargs.
     """
     # Add _ray_trace_ctx to function signature
-    if not _is_tracing_enabled():
-        return function
```

Comment:
This is the main change: if `ray.init` hasn't been called then `_is_tracing_enabled` isn't set properly yet.

Need to always inject this `kwarg` optimistically. This shouldn't affect anything unless users have a kwarg called `_ray_trace_ctx` in their function.

Note that this is already how it worked for actor methods.